Clarify canonical format used for Breadcrumbs, Exceptions and Threads (#1554)
getsentry/sentry-docs: src/collections/_documentation/development/sdk-dev/event-payloads/index.md

sidebar_order: 3
---

> Events are the fundamental data that clients, often through use of an SDK, send

Reviewer: ```suggestion
Events are the fundamental data that clients, often through the use of an SDK, send
```

> to the Sentry server.

Author: I realized we never formally define "event" in the documentation.
Started with something simple.

In `src/collections/_documentation/workflow/search.md` we say:

> Events are the underlying event data captured using Sentry SDKs (read: errors and exceptions).

Though that is not precise because there is `capture_message` and perf metrics in spans that are not errors/exceptions.

The API endpoint on the Sentry server where event payloads are sent is
`/api/{PROJECT_ID}/store/`.

{% capture __alert_content -%}

> We strive to document the canonical format of events and its additional

Reviewer: ```suggestion
We strive to document the canonical format of an event and its additional
```

> interfaces. However, note that for backwards compatibility the server also

Reviewer: ```suggestion
interfaces. However, for backwards compatibility the server also
```

understands events that are not in the canonical format described throughout the
documentation.

Existing SDKs may be using a historical format that is not recommended for new

> code.

Author: I want this to be able to explain to people/myself when we see SDK code that doesn't match the documentation.

{%- endcapture -%}

{%- include components/alert.html title="Note on backwards compatibility" content=__alert_content level="info"%}

{% capture __alert_content -%}
If documentation is lacking or outdated, please let us know by [opening an
issue](https://github.com/getsentry/sentry-docs/issues/new).

SDK developers might benefit from consulting the [source code defining the
protocol as understood by the

> server](https://github.com/getsentry/relay/tree/master/relay-general/src/protocol).

Author: This would have saved me a lot of time in the beginning.

Given that docs are not perfect, I don't see a reason to not link to it here as a stop-gap solution. Ideally the source code would be well documented and we could generate the API docs.

**STEP BY STEP**

Reviewer 2: Alright. Just beware that this link is not stable and the code will be moved in near term.

Reviewer 2: I will make sure to update this link here if we move the code.

Author: Right we can't guarantee it will be there forever. In this case it wouldn't be useful to have a link to a commit/tag/branch/[`@{2020.03.18}`](https://github.com/getsentry/relay/tree/@%7B2020.03.18%7D/relay-general/src/protocol) because it would be outdated even earlier than the code moving.
